src/sbus/interface/sbus_interface.c

```diff
@@ -348,6 +348,7 @@ sbus_listener_copy(TALLOC_CTX *mem_ctx,
 
     /* All data is either pointer to a static data or it is not a pointer.
      * We can just copy it. */
+    /* coverity[store_writes_const_field] */
```

Comment:
Can we just replace alloc+copy with single `talloc_memdup()`?

Reply:
Btw, where could I read about `store_writes_const_field`?

Reply:
I didn't find any direct information about this. However, the error itself has some information:

Error: WRITE_CONST_FIELD ([CWE-843](https://cwe.mitre.org/data/definitions/843.html)): [[#def115]](https://cov01.lab.eng.brq2.redhat.com/covscanhub/task/279726/log/sssd-pr6686-01972.fc37/scan-results.html#def115)
sssd-pr6686/src/sbus/sbus_interface_declarations.h:276: field_definition: **A const-qualified field is defined.**
sssd-pr6686/src/sbus/interface/sbus_interface.c:351: store_writes_const_field: **A write to an aggregate overwrites a const-qualified field within the aggregate.**
```
#  349|       /* All data is either pointer to a static data or it is not a pointer.
#  350|        * We can just copy it. */
#  351|->     memcpy(copy, input, sizeof(struct sbus_listener));
#  352|   
#  353|       return copy;
```

My understanding is that the reported error is that we are writing with `memcpy()` to the structure `struct sbus_listener` which includes two `const struct sbus_invoker` members. Which is true, but is also right.

Reply:
Using `talloc_memdup()` seems to be a good idea. Let me check that also to confirm coverity doesn't complain.

Reply:
**Update2:** Coverity didn't complain, so I'm adopting it.

Reply:
I think "COVERITY: Ignore a WRITE_CONST_FIELD warning" commit message is now a bit misleading.
You could mention this change "resolves" this warning, but imo it's not "ignore" anymore.
Otherwise ACK.

Reply:
LOL, I was doing exactly that change when I received your comment.